lib/index.js

```diff
@@ -520,103 +424,96 @@ internals.diffColumns = (migrationGroup, joinTables, mode, cb) => {
+                            break;
 
-                                if (mode !== 'alter') {
-                                    return;
-                                }
+                        case 'alter':
 
-                                const db2KnexKey0 = Mappings.convertFuncs.db2Knex(delta[key][0]);
-                                const db2KnexKey1 = Mappings.convertFuncs.db2Knex(delta[key][1]);
+                            // Enforcing 'mode' here
+                            if (mode !== 'alter') {
+                                return;
+                            }
 
-                                modelDeltaObject.alter[key] = [db2KnexKey0, db2KnexKey1];
+                            const db2KnexKey0 = Mappings.convertFuncs.db2Knex(delta[key][0]);
+                            const db2KnexKey1 = Mappings.convertFuncs.db2Knex(delta[key][1]);
 
-                                break;
-                            case 'drop':
+                            modelDeltaObject.alter[key] = [db2KnexKey0, db2KnexKey1];
+                            break;
 
-                                if (mode !== 'alter') {
-                                    return;
-                                }
+                        case 'drop':
 
-                                const db2KnexKey = Mappings.convertFuncs.db2Knex(dbSchema[key]);
+                            // Enforcing 'mode' here
+                            if (mode !== 'alter') {
+                                return;
+                            }
 
-                                modelDeltaObject.drop.push({
-                                    columnName: key,
-                                    columnType: db2KnexKey
-                                });
+                            const db2KnexKey = Mappings.convertFuncs.db2Knex(dbSchema[key]);
 
-                                break;
-                        }
-                    });
+                            modelDeltaObject.drop.push({
+                                columnName: key,
+                                columnType: db2KnexKey
+                            });
+                            break;
+                    }
+                });
 
-                    knexColumnDeltaObject.changes.push(modelDeltaObject);
-                }
+                knexColumnDeltaObject.changes.push(modelDeltaObject);
+            }
 
-                modelNext();
-            });
+            next();
+        });
     },
-    (_) => {
+    (ignoreError) => {
 
-        return cb(_, knexColumnDeltaObject);
+        return cb(ignoreError, knexColumnDeltaObject);
     });
 };
 
 
-internals.getColumnsForTable = (knex, tableName) => {
-
-    let resolve;
-
-    const p = new Promise((res, rej) => {
-
-        resolve = res;
-    });
+internals.getColumnsForTable = (knex, tableName, cb) => {
 
     knex.table(tableName).columnInfo()
         .then((dbSchema) => {
```

Comment:
More promises mixing in there.

Reply:
In this case an error could get swallowed, e.g. if `columnInfo()` fails, since there's no `catch()`.